deployment/cdk/opensearch-service-migration/README.md

```diff
@@ -1,184 +1,187 @@
 
-1- Install the **CDK CLI** tool by running:
-```
-npm install -g aws-cdk
+5- There is a known issue where service linked roles fail to get applied when deploying certain AWS services for the first time in an account. This can be resolved by simply deploying again (for each failing role) or avoided entirely by creating the service linked role initially like seen below:
+```shell
+aws iam create-service-linked-role --aws-service-name opensearchservice.amazonaws.com && aws iam create-service-linked-role --aws-service-name ecs.amazonaws.com
 ```
 
-2- Configure the desired **[AWS credentials](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html#getting_started_prerequisites)**, as these will dictate the region and account used for deployment.
+### First time using CDK in this region?
 
-3- **Bootstrap CDK**: if you have not run CDK previously in the configured region of you account, it is necessary to run the following command to set up a small CloudFormation stack of resources that CDK needs to function within your account
+If this is your first experience with CDK, follow the steps below to get started:
 
+1- Install the **CDK CLI** tool, if you haven't already, by running:
+```shell
+npm install -g aws-cdk
 ```
-cdk bootstrap
+
+2- **Bootstrap CDK**: if you have not run CDK previously in the configured region of you account, it is necessary to run the following command from the `opensearch-service-migration` directory to set up a small CloudFormation stack of resources that CDK needs to function within your account
+
+```shell
+# Execute from the deployment/cdk/opensearch-service-migration directory
+cdk bootstrap --c contextId=demo-deploy
 ```
 
 Further CDK documentation [here](https://docs.aws.amazon.com/cdk/v2/guide/cli.html)
 
-### Deploying your CDK
-Before deploying your CDK you should fill in any desired context parameters that will dictate the composition of your OpenSearch Service Domain
+## Deploying the CDK
+This project uses CDK context parameters to configure deployments. These context values will dictate the composition of your stacks as well as which stacks get deployed.
+
+The full list of available configuration options for this project are listed [here](./options.md). Each option can be provided as an empty string `""` or simply not included, and in each of these 'empty' cases the option will use the project default value (if it exists) or CloudFormation's default value.
 
-This can be accomplished by providing these options in a `cdk.context.json` file and then deploying like so:
+A set of demo context values (using the `demo-deploy` label) has been set in the `cdk.context.json` located in this directory, which can be customized or used as is for a quickstart demo solution.
+
+This demo solution can be deployed with the following command:
+```shell
+cdk deploy "*" --c contextId=demo-deploy --require-approval never --concurrency 3
```

Comment:
From my own experience I have never felt the need for more than 3. It is really the minimum of two that you want so that we are able to achieve deploying MSK and a Domain side by side. Past that our migration services have dependencies on MSK or possibly on each other which would stop more than 3 as well.